h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -609,14 +614,49 @@ private void optionallySplitTrainingDataset() {
 
   private DistributionFamily inferDistribution(Vec response) {
     int numOfDomains = response.domain() == null ? 0 : response.domain().length;
-    if (numOfDomains == 0)
-      return DistributionFamily.gaussian;
-    if (numOfDomains == 2)
-      return DistributionFamily.bernoulli;
-    if (numOfDomains > 2)
-      return DistributionFamily.multinomial;
-
-    throw new RuntimeException("Number of domains is equal to 1.");
+    if (_buildSpec.build_control.distribution == DistributionFamily.AUTO) {
+      if (numOfDomains == 0)
+        return DistributionFamily.gaussian;
+      if (numOfDomains == 2)
+        return DistributionFamily.bernoulli;
+      if (numOfDomains > 2)
+        return DistributionFamily.multinomial;
+
+      throw new RuntimeException("Number of domains is equal to 1.");
```

Comment:
@sebhrusen We actually use the term "domain" in the model code (it's exposed in the model JSON in Python, but not stored in R models for some reason).  I actually think we should expose this in R too because it's handy to have this info sometimes.  However, if we want to use this idea in an error message, I might use a different, more obvious term like "number of classes".

Reply:
Changed to `"Number of classes is equal to 1."`. And in the following messages changed to use "classes" instead of "domains".